client/widmgr/widmgr.go

```diff
@@ -186,8 +195,37 @@ func (m *WIDMgr) Shutdown() {
 	}
 }
 
-// getIdentities fetches all signed identities or returns an error.
-func (m *WIDMgr) getIdentities() error {
+// restoreStoredIdentities recreates the state of the WIDMgr from a previously
+// saved state, so that we can avoid asking for all identities again after a
+// client agent restart. It returns true if the caller should immediately call
+// getIdentities because one or more of the identities is expired.
```

Comment:
Would re-fetching all identities (even if not expired) caused a task to restart? I think we've seen cases where tasks are unexpectedly restarted, so it would be good to avoid them here as well.

Reply:
We don't have any kind of change mode on identities (yet), so no this won't cause tasks to restart even in this restore path. The `identity_hook` will write new JWTs and expose them via hook resources and that's it. It's up to the application to detect that (if it needs to) for now.